Framework/Algorithms/src/FilterEvents.cpp

```diff
@@ -447,19 +447,19 @@ void FilterEvents::processMatrixSplitterWorkspace() {
   // Check input workspace validity
   assert(m_matrixSplitterWS);
 
-  const MantidVec &vecX = m_matrixSplitterWS->readX(0);
-  const MantidVec &vecY = m_matrixSplitterWS->readY(0);
-  size_t sizex = vecX.size();
-  size_t sizey = vecY.size();
+  auto &X = m_matrixSplitterWS->x(0);
+  auto &Y = m_matrixSplitterWS->y(0);
+  size_t sizex = X.size();
+  size_t sizey = Y.size();
   assert(sizex - sizey == 1);
```

Comment:
Tests like this are now not really needed any more. There are two options for getting rid of them:
- In some cases, if the test is only for the input, using `binEdges()` instead of `x()` might be the right thing.
- In other cases there should probably be an algorithm property validator (`HistogramValidator`) or a check of `MatrixWorkspace::isHistogram`.